cmd/zfs/zfs_main.c

```diff
@@ -7237,15 +7256,27 @@ unshare_unmount(int op, int argc, char **argv)
 
 			switch (op) {
 			case OP_SHARE:
+#if defined(__FreeBSD__)
+				if (zfs_unshareall_bypath(node->un_zhp,
+				    node->un_mountp) != 0)
+					ret = 1;
+#else
 				if (zfs_unshareall_bytype(node->un_zhp,
 				    node->un_mountp, protocol) != 0)
 					ret = 1;
+#endif
 				break;
 
 			case OP_MOUNT:
+#ifdef __FreeBSD__
+				if (zfs_unmount(node->un_zhp,
+				    NULL, flags) != 0)
```

Comment:
This feel a bit heavy handed and doesn't look right.  I see that on FreeBSD and illumos `node->un_mountp` is passed here instead of NULL.

Reply:
that's still not the same thing that Linux does

Reply:
Doing that for both breaks rollback_003 on linux - is there actually a solution that works on both?